frontend/src/pages/QuestionPage.js

```diff
@@ -1,345 +1,417 @@
+		setOpen(true);
+	};
+	// eslint-disable-next-line no-unused-vars
+	const reportCreator = (reportType, otherDescription) => {
+		let body = '';
+		if (otherDescription !== null) {
+			body = { question: id, type: reportType, body: otherDescription };
+		} else {
+			body = { question: id, type: reportType };
+		}
+		createReport(
+			body,
+			(res) => {
+				enqueueSnackbar(
+					'Foi criado com sucesso a sua denúncia com id:' + String(res.data.id),
+					{ variant: 'success' }
+				);
+			},
+			() => {
+				enqueueSnackbar('Não foi possível criar a sua denúncia, tente mais tarde...', {
+					variant: 'error',
 				});
-			});
-		} else this.removeUpvote();
-	}
+			}
+		);
+	};
+	const onClose = (reportType, otherDescription) => {
+		// We only want to submit stuff when the user chooses a report Type, and in the case of the Other there must be some description
+		// If there is some description regardless of report type it must have at least 30 characters
+		setOpen(false);
 
-	removeUpvote() {
-		removeUpvoteAPI(this.props.data.id, (res) => {
-			this.setState({
-				upvoted: false,
-				votes: res.data.votes,
+		if (reportType === 'Other' && otherDescription === null) {
+			enqueueSnackbar(
+				"Submissão Inválida! Se escolheu 'Outro...', precisa de clarificar o problema encontrado...",
+				{
+					variant: 'error',
+				}
+			);
+		} else if (
+			reportType !== null &&
+			otherDescription !== null &&
+			String(otherDescription).length < 30
+		) {
+			enqueueSnackbar('A sua descrição necessita de um mínimo de 30 caractéres', {
+				variant: 'error',
 			});
-		});
-	}
-
-	downvote() {
-		if (!this.state.downvoted) {
-			downvoteAPI(this.props.data.id, (res) => {
-				this.setState({
-					downvoted: true,
-					upvoted: false,
-					votes: res.data.votes,
-				});
+			return;
+		} else if (reportType !== null && isSwear(otherDescription)) {
+			enqueueSnackbar('A sua descrição contém expressões agressivas/impróprias', {
+				variant: 'warning',
 			});
-		} else this.removeDownvote();
-	}
+			return;
+		} else if (reportType !== null) {
+			reportCreator(reportType, otherDescription);
+		}
+	};
+	useEffect(() => {
+		if (question) {
+			sortRerender(comments);
+		} else {
+			fetchQuestion(id, (res) => {
+				setQuestion(res.data);
+				sortRerender(res.data.comment);
+			});
+		}
+	}, [sortingType]);
 
-	removeDownvote() {
-		removeDownvoteAPI(this.props.data.id, (res) => {
-			this.setState({
-				downvoted: false,
-				votes: res.data.votes,
+	useEffect(() => {
+		getUser((res1) => {
+			if (res1.data?.mod) {
+				setIsUserMod(true);
+			}
+
+			getProfile(res1.data.profile, (res2) => {
+				setUserID(res2.data.user.id);
 			});
 		});
-	}
+	}, []);
 
-	render() {
-		return (
-			<div style={{ padding: 14 }}>
-				<Grid container wrap='nowrap' spacing={2}>
-					<Grid item>
-						<AvatarUser user={this.props.data.author} />
-						<Typography variant={'h6'}>{this.state.votes}</Typography>
+	if (question === null) return <></>;
```

Comment:
There's the `Loading` component that can be used here.

Reply:
@JeronimoMendes  Done